### DISCUSSION TITLE
How to map a merged branch (with a ticket id) to an environment. Marry Vercel, Github, and Jira integrations.

### DISCUSSION THREAD

**Discussion by natezander** [atlassian/github-for-jira#1947] [Q&A]
**As a director, I need to know which environments a specific ticket has been deployed to.**

**Vercel** has the concept of environments, and we successfully create deployments w/ environment variables based on branch names.

![Screenshot 2023-03-07 at 11 44 25 AM](https://user-images.githubusercontent.com/5290498/223523677-3ef51018-026a-4b74-9501-d3bd85df4f02.png)

**Github** has the concept of [environments](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment), but I have never been able to figure out how to map them to a branch pattern or to a vercel deploy. I thought I found the holy grail with the recently released [environment mapping feature in GH](https://github.com/atlassian/github-for-jira/blob/main/docs/deployments.md#environment-mapping), but this seems tied to their own concept of environment variables and GH actions. 

And **Jira** wraps this all together by publishing branch and environments for each ticket.

![Screenshot 2023-03-07 at 11 46 53 AM](https://user-images.githubusercontent.com/5290498/223523781-7e48b6d9-600f-4a93-9190-590652fd10ce.png)
![Screenshot 2023-03-07 at 11 47 00 AM](https://user-images.githubusercontent.com/5290498/223523783-2b9f90ef-871d-4fc7-9d4b-ebbc2343d960.png)
![Screenshot 2023-03-07 at 11 49 59 AM](https://user-images.githubusercontent.com/5290498/223523784-143df94b-3e0f-4474-8a7c-57ff42abfe70.png)

Has anyone successfully linked these three products together? 

My ultimate goal is to determine 'if a ticket has been merged in to the `development` **branch**, then it is live in the `development` **environment**. When those tickets have been merged into the `staging` branch, then those tickets are live in the `staging` environment. You can see that most deployments are simply listed as `Preview` or `Undefined`

It does appear that Jira/GH/Vercel properly matches pushes to the `master` branch and marks them as `production` environment.

**Comment by DanielKozij** [comment]
Are there any new updates related to this?
